lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -102,6 +102,13 @@ static CodableConformanceType typeConformsToCodable(DeclContext *context,
 /// \param proto The \c ProtocolDecl to check conformance to.
 static CodableConformanceType
 varConformsToCodable(DeclContext *DC, VarDecl *varDecl, ProtocolDecl *proto) {
+  // CodingKeys enum that couldn't be generated correctly because of some other
+  // error elsewhere. e.g. Invalid redeclaration of members. So let's fail early
+  // here as type not validated.
+  if (varDecl->getValueInterfaceType()->hasError()) {
```

Comment:
We use `varDecl->getValueInterfaceType()` later on too, so perhaps you can extract it into a local variable which is used here and below.

Reply:
Fixed! Thanks